storyblok/field-plugin
# EXT-1287: add a script to bump versions of either field-plugin or cli

## What?

This PR adds a script to deploy `@storyblok/field-plugin` or `@storyblok/field-plugin-cli`.

## Why?

There are routine works to deploy either libraries, and it'd be nice to have an automated way, so that we don't have to think about branch names, pull-request titles, etc.

## How to test? (optional)

`yarn deploy` is supposed to let you update the version and create a pull request for you, but it doesn't work in this branch (only in `main`).

So in order to test it, you need to comment out the following lines:

```js
// Check if the current branch is `main`
const currentBranch = await $`git branch --show-current`
if (currentBranch.toString().trim() !== 'main') {
  print(bold(red('[Error]')), 'This command runs only from `main` branch.')
  exit(1)
}

// Check if the working directory is clean
try {
  await $`git diff-index --quiet HEAD --`
} catch (processOutput) {
  if (processOutput.exitCode === 1) {
    print(
      bold(red('[Error]')),
      'There are uncommitted changes in the working directory. Please clean them up before proceeding.',
    )
    exit(1)
  }
}
```

And you may also want to comment out the last lines:

```js
await $`git push -u origin ${branchName}`
await $`gh pr create --fill`
await $`gh pr view --web`
```

After this, you can run `yarn deploy` at the root of this repository.

---

It works something like this:


https://user-images.githubusercontent.com/499898/230639697-a256a045-b4e2-440d-b5d3-b7978fd4ca45.mp4
